test/e2e/fail_forward_e2e_test.go

```diff
@@ -102,40 +103,30 @@ var _ = Describe("Fail Forward Upgrades", func() {
 			By("updating the catalog with a broken v0.2.0 bundle image")
 			brokenProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, failForwardTestDataBaseDir, "example-operator.v0.2.0.yaml"))
 			Expect(err).To(BeNil())
-
 			err = magicCatalog.UpdateCatalog(context.Background(), brokenProvider)
 			Expect(err).To(BeNil())
 
-			By("verifying the subscription is referencing a new installplan")
-			subscription, err = fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasInstallPlanDifferentChecker(originalInstallPlanRef.Name))
-			Expect(err).Should(BeNil())
-
-			By("waiting for the bad InstallPlan to report a failed installation state")
-			ref := subscription.Status.InstallPlanRef
-			_, err = fetchInstallPlan(GinkgoT(), crclient, ref.Name, ref.Namespace, buildInstallPlanPhaseCheckFunc(operatorsv1alpha1.InstallPlanPhaseFailed))
-			Expect(err).To(BeNil())
+			By("waiting for the subscription to maintain the example-operator.v0.1.0 status.currentCSV")
+			Consistently(subscriptionCurrentCSVGetter(crclient, subscription.GetNamespace(), subscription.GetName())).Should(Equal("example-operator.v0.1.0"))
 
+			By("verifying the subscription is referencing the same InstallPlan")
+			subscription, err = fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasInstallPlanChecker)
+			Expect(err).Should(BeNil())
+			Expect(subscription.Status.InstallPlanRef.Name).To(Equal(originalInstallPlanRef.Name))
 		})
 		AfterEach(func() {
 			By("removing the testing catalog resources")
 			Expect(magicCatalog.UndeployCatalog(context.Background())).To(BeNil())
 		})
-		It("eventually reports a successful state when multiple bad versions are rolled forward", func() {
+		It("eventually gets created and reports a successful state when multiple bad versions are rolled forward", func() {
 			By("patching the catalog with another bad bundle version")
 			badProvider, err := NewFileBasedFiledBasedCatalogProvider(filepath.Join(testdataDir, "fail-forward/multiple-bad-versions", "example-operator.v0.2.1.yaml"))
 			Expect(err).To(BeNil())
-
 			err = magicCatalog.UpdateCatalog(context.Background(), badProvider)
 			Expect(err).To(BeNil())
 
-			By("waiting for the subscription to have the example-operator.v0.2.1 status.updatedCSV")
-			subscription, err = fetchSubscription(crclient, subscription.GetNamespace(), subscription.GetName(), subscriptionHasCurrentCSV("example-operator.v0.2.1"))
-			Expect(err).Should(BeNil())
-
-			By("waiting for the bad v0.2.1 InstallPlan to report a failed installation state")
-			ref := subscription.Status.InstallPlanRef
-			_, err = fetchInstallPlan(GinkgoT(), crclient, ref.Name, ref.Namespace, buildInstallPlanPhaseCheckFunc(operatorsv1alpha1.InstallPlanPhaseFailed))
-			Expect(err).To(BeNil())
```

Comment:
Actually, I'm having second thoughts on this. How I rearanged tests to be "when InstallPlan is not being created due to failed upgrade" just provides extra coverage for what we cover elsewhere (in `test/e2e/subscription_e2e_test.go` and and extra tests defined in #2961).

> This should test our ability to recover from a failed installPlan, so it may make sense to remove fail-forward tests around missing bundle images.

I think this is the right approach.

> shouldn't we verify the failed installPlan state as a prerequisite for testing fail-forward?

I'll explore getting back to "an InstallPlan is reporting a failed state" and emulating failed `InstallPlan` execution somewhere in `BeforeEach`.